### PR TITLE
update titiler-pgstac and use PgSTACReader for the item endoints

### PR DESCRIPTION
**Pull request by vincentsarago**
## Description

This PR is like https://github.com/microsoft/planetary-computer-apis/pull/55 but re-use code from titiler-pgstac

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

🤷 

## Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review
- [ ] Changelog has been updated
- [ ] Documentation has been updated
- [x] Unit tests pass locally (./scripts/test)
- [x] Code is linted and styled (./scripts/format)

### CODE REVIEW

File: pctiler/pctiler/endpoints/item.py [hunk 1]
```diff
@@ -24,21 +24,6 @@
...
-    collection: str = Query(..., description="STAC Collection ID"),
-    item: str = Query(..., description="STAC Item ID"),
-) -> str:
-    return urljoin(
-        get_settings().stac_api_url, f"collections/{collection}/items/{item}"
-    )
-
-
-@dataclass
-class MapParams:
```
Comment (vincentsarago) [inline pctiler/pctiler/endpoints/item.py:37]: this wasn't used 🤷‍♂️

File: pctiler/pctiler/reader.py [hunk 2]
```diff
@@ -50,13 +49,13 @@ def _get_asset_url(self, asset: str) -> str:
 
 
 @attr.s
-class CustomSTACReader(MultiBaseReader):
+class MosaicSTACReader(MultiBaseReader):
```
Comment (vincentsarago) [inline pctiler/pctiler/reader.py:52]: rename this to avoid confusion with the `ItemSTACReader`

**Comment by vincentsarago** [comment]
@moradology I hope you don't mind I did a second try as what you were doing in #55 but using new code added in titiler-pgstac.

File: pctiler/pctiler/reader.py [hunk 3]
```diff
@@ -33,7 +32,7 @@ def get_cache_key(
 
 
 @attr.s
-class ItemSTACReader(STACReader):
+class ItemSTACReader(PgSTACReader):
```
Comment (mmcfarland) [inline pctiler/pctiler/reader.py:35]: Looks like the Item property on `ItemSTACReader` is now `input`, so the `_get_asset_url` implementation in this class needs to be updated e.g., `self.input.collection_id` to check the render configs.

**Review by mmcfarland — COMMENTED** [review]
Hit a snag with changed property names mentioned below, but otherwise looking good.

File: pctiler/pctiler/reader.py (hunk 3 above)
Comment (vincentsarago) [reply pctiler/pctiler/reader.py:35]: Good Catch 🙏

**Comment by vincentsarago** [comment]
# TODO 
- [ ] add tests

**Comment by vincentsarago** [comment]
Last commit refactored the test suite for pctiler to make sure we were running test `async`

**Comment by moradology** [comment]
Looks good. I'm happy to see some of the functionality that would have been implemented here found a home, upstream, in `titiler-pgstac`. Async test fixes were also a win. 👍

**Review by mmcfarland — APPROVED** [review]

**Pull request merged**